feature/experimental/tunnel/otg_tests/tunnel_interface_based_ipv4_gre_encapsulation_test/tunnel_interface_based_ipv4_gre_encapsulation_test.go

```diff
@@ -0,0 +1,482 @@
+		Name:    "otgdst2",
+		IPv4:    "198.18.4.2",
+		IPv4Len: interfacePlen4,
+		MAC:     "00:00:03:03:03:03",
+	}
+)
+
+func TestMain(m *testing.M) {
+	fptest.RunTests(m)
+}
+
+func TestTunnelEncapsulationByGREOverIPv4WithLoadBalance(t *testing.T) {
+	dut := ondatra.DUT(t, "dut")
+	dutPort1 := dut.Port(t, "port1")
+	dutPort2 := dut.Port(t, "port2")
+	dutPort3 := dut.Port(t, "port3")
+	ate := ondatra.ATE(t, "ate")
+	ateport1 := ate.Port(t, "port1")
+	ateport2 := ate.Port(t, "port2")
+	ateport3 := ate.Port(t, "port3")
+	egressInterfaces := []string{dutPort2.Name(), dutPort3.Name()}
+	initialEgressPkts := make([]uint64, tunnelCount)
+	initialTunnelInPkts := make([]uint64, tunnelCount)
+	initialTunnelOutPkts := make([]uint64, tunnelCount)
+	tunnelLoadblanceDiff := tunnelCount * 3
+	interfaceLoadblanceDiff := tolerance
+	t.Run("Configure dut with 32 tunnel interface with one ingress and 2 egress interface", func(t *testing.T) {
+		configureTunnelBaseOnDUT(t, dut, dutPort1, &dutIntf1)
+		configureTunnelBaseOnDUT(t, dut, dutPort2, &dutIntf2)
+		configureTunnelBaseOnDUT(t, dut, dutPort3, &dutIntf3)
+		step := 0
+		var overlayIPv4Nh []string
+		for unit := 0; unit < tunnelCount; unit++ {
+			tunnelSrc := incrementAddress(t, tunnelSrcIpv4Network, unit, "host")
+			tunnelDstNetwork := incrementAddress(t, tunnelDesIpv4Network, unit, "network")
+			tunnelDst := incrementAddress(t, tunnelDstNetwork, 1, "host")
+			tunnelIpv4address := incrementAddress(t, tunnelNhIpv4Network, step, "host")
+			t.Logf("unit : %d tunnel ipv4 address: %s/%d  tunnel source address: %s tunnel destination: %s", unit, tunnelIpv4address, tunnelPlen4, tunnelSrc, tunnelDst)
+			if deviations.TunnelConfigPathUnsupported(dut) {
+				configureTunnelInterface(t, tunnelInterface, unit, tunnelSrc, tunnelDst, tunnelIpv4address, tunnelPlen4, dut)
+			}
+			overlayIPv4Nh = append(overlayIPv4Nh, incrementAddress(t, tunnelIpv4address, 1, "host"))
+			step = step + 2
+		}
+
+		t.Logf("Configure routing instance on dut")
+		configureNetworkInstance(t, dut)
+		t.Logf("Configure IPv4 tunnel destination address reachable via ECMP link")
+		underlayIpv4Nh := []string{otgIntf2.IPv4, otgIntf3.IPv4}
+		for i, nextHop := range underlayIpv4Nh {
+			_, ipv4Destination := fetchNetworkAddress(t, tunnelDesIpv4Network, 19)
+			t.Logf("configuring static route in %s destination %s with next-hop %s", dut, ipv4Destination, nextHop)
+			configIPv4StaticRoute(t, dut, ipv4Destination, nextHop, strconv.Itoa(i))
+		}
+		t.Logf("overlay static route via tunnel for an original IPv4 destination prefix")
+		for i, nextHop := range overlayIPv4Nh {
+			_, ipv4Destination := fetchNetworkAddress(t, encapInnerDesIpv4Network, interfacePlen4)
+			t.Logf("configuring static route in %s destination %s with next-hop %s", dut, ipv4Destination, nextHop)
+			configIPv4StaticRoute(t, dut, ipv4Destination, nextHop, strconv.Itoa(i))
+		}
+	})
+	t.Run("Configure OTG ports", func(t *testing.T) {
+		top := ate.OTG().NewConfig(t)
```

Comment:
this function is deprecated. Just use gosnappi.NewConfig()

Reply:
Modified to gosnappi.NewConfig()